File: src/cz/vutbr/fit/interlockSim/objects/cells/NodeCell.java
/* Brno University of Technology
 * Faculty of Information Technology
 * 
 * BSc Thesis	2006/2007
 * 
 * Railway Interlocking Simulator
 * 
 * Bedrich Hovorka
 */
package cz.vutbr.fit.interlockSim.objects.cells;

import cz.vutbr.fit.interlockSim.objects.paths.PathSeparator;

/**
 * Node managed by dispatcher (if is in main graph)
 *
 */
public abstract class NodeCell extends AbstractCell implements PathSeparator { 
	private final SpatialType spatialType;
	private String toString = "";
	private String name = "";
	
	/**
      * @param spatialType
      */
	public NodeCell(SpatialType spatialType) {
		this.spatialType = spatialType;
	}
	
	public SpatialType getSpatialType() {
		return spatialType;
	}
	
	/**
	 * setter
	 * @param name
	 */
	public void setName(String name) {
		this.name = name;
		toString = getClass().getSimpleName() + ':' + String.valueOf(name);
	}

	/**
	 * 
	 * @return
	 */
	public String getName() {
		return name;
	}
	
	@Override
	public String toString() {
		return toString;
	}
}

File: src/main/kotlin/cz/vutbr/fit/interlockSim/objects/cells/NodeCell.kt
/* Brno University of Technology
 * Faculty of Information Technology
 *
 * BSc Thesis  2006/2007
 *
 * Railway Interlocking Simulator
 *
 * Bedrich Hovorka
 */
package cz.vutbr.fit.interlockSim.objects.cells

import cz.vutbr.fit.interlockSim.objects.core.Cell

import cz.vutbr.fit.interlockSim.objects.core.PathSeparator

/**
 * Node managed by dispatcher (if is in main graph)
 *
 */
abstract class NodeCell(
	private val spatialType: Cell.SpatialType
) : AbstractCell(),
	PathSeparator {
	private var toStringValue: String = ""
	private var name: String = ""

	override fun getSpatialType(): Cell.SpatialType = spatialType

	/**
	 * setter
	 * @param name
	 */
	open fun setName(name: String) {
		this.name = name
		toStringValue = this::class.java.simpleName + ':' + name
	}

	/**
	 *
	 * @return
	 */
	open fun getName(): String = name

	override fun toString(): String = toStringValue
}
